internal/reconciler/logpipeline/reconciler.go

```diff
@@ -86,32 +85,37 @@ type IstioStatusChecker interface {
 type Reconciler struct {
 	client.Client
 
-	config             Config
-	prober             DaemonSetProber
+	config *Config
+	syncer syncer
+
+	// Dependencies
+	agentProber        DaemonSetProber
 	flowHealthProber   FlowHealthProber
-	tlsCertValidator   TLSCertValidator
-	syncer             syncer
-	overridesHandler   OverridesHandler
 	istioStatusChecker IstioStatusChecker
+	overridesHandler   OverridesHandler
+	tlsCertValidator   TLSCertValidator
 }
 
-func NewReconciler(
+func New(
 	client client.Client,
-	config Config,
+	config *Config,
```

Comment:
Why are we changing the config field here to be a pointer? 
Is it to make it more efficient and pass a pointer instead of the entire Config struct?
Maybe we make it consistent with the [config](https://github.com/kyma-project/telemetry-manager/blob/ee2ec606a42648fe71385d511d126a9f23b5e210/internal/reconciler/logparser/reconciler.go#L54) field in the Logparser reconciler and either make both of them pointers or just structs?